backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Saldobilanz_Report.sql

```diff
@@ -1,176 +1,564 @@
+           SUM(CASE WHEN ParentValue2 IS NOT NULL THEN SameYearSum ELSE NULL END) OVER ( PARTITION BY ParentValue2) AS L2_SameYearSum,
+           SUM(CASE WHEN ParentValue2 IS NOT NULL THEN LastYearSum ELSE NULL END) OVER ( PARTITION BY ParentValue2) AS L2_LastYearSum,
+           SUM(CASE WHEN ParentValue1 IS NOT NULL THEN SameYearSum ELSE NULL END) OVER ( PARTITION BY ParentValue1) AS L1_SameYearSum,
+           SUM(CASE WHEN ParentValue1 IS NOT NULL THEN LastYearSum ELSE NULL END) OVER ( PARTITION BY ParentValue1) AS L1_LastYearSum,
+
+           -- More info:
+           C_Calendar_ID
+            ,
+           C_ElementValue_ID,
+
+           p_ad_org_id                                                                                              AS ad_org_id,
+           a.iso_code
+    FROM (
+             SELECT tb.ParentName1
+                  , tb.ParentValue1
+                  , tb.ParentName2
+                  , tb.ParentValue2
+                  , tb.ParentName3
+                  , tb.ParentValue3
+                  , tb.ParentValue4
+                  , tb.ParentName4
+                  , tb.Name
+                  , tb.Value
+                  , tb.AccountType
+
+                  , tb.SameYearSum
+                  , tb.LastYearSum
+
+
+                  , tb.IsConvertToEUR
+
+                  --
+                  , v_AcctSchemaInfo.C_Currency_ID -- Accounting currency
+                  , v_AcctSchemaInfo.AD_Client_ID
 
+                  , v_AcctSchemaInfo.C_Calendar_ID
+                  , tb.C_ElementValue_ID
+                  , v_AcctSchemaInfo.iso_code
+             FROM tmp_balances tb
+         ) a;
+
+    GET DIAGNOSTICS v_rowcount = ROW_COUNT;
+    RAISE NOTICE 'Created tmp_reports with % rows', v_rowcount;
+
+
+    IF p_IsShowProductDetails = 'Y' OR p_IsShowActivityDetails = 'Y' THEN
+
+        -- compute balances per activity and product
+        DROP TABLE IF EXISTS tmp_Fact_Acct_Per_DateAcct;
+        CREATE TEMPORARY TABLE tmp_Fact_Acct_Per_DateAcct AS
+        SELECT fa.AD_Client_ID
+             , fa.Account_ID
+             , fa.c_activity_id
+             , fa.m_product_id
+             , fa.C_AcctSchema_ID
+             , fa.PostingType
+             , p.C_Period_ID
+             , p.C_Year_ID
+             , fa.DateAcct
+             -- Aggregated amounts
+             , COALESCE(SUM(AmtAcctDr), 0) AS AmtAcctDr
+             , COALESCE(SUM(AmtAcctCr), 0) AS AmtAcctCr
+             , COALESCE(SUM(Qty), 0)       AS Qty
+        FROM Fact_Acct fa
+                 LEFT OUTER JOIN C_Period p ON (p.C_Period_ID = fa.C_Period_ID)
+        WHERE fa.dateacct <= p_date
+          AND fa.ad_org_id = p_ad_org_id
+        GROUP BY fa.AD_Client_ID
+               , p.C_Period_ID
+               , fa.DateAcct
+               , fa.C_AcctSchema_ID
+               , fa.PostingType
+               , fa.Account_ID
+               , fa.c_activity_id
```

Comment:
use 
SELECT 
(case when p_IsShowActivityDetails='Y' then fa.c_activity_id end) as c_activity_id
....
GROUP BY
....
(case when p_IsShowActivityDetails='Y' then fa.c_activity_id end)


and same for prod